### PR TITLE
reference/performance: add doc for join reorder

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add a doc for the Join Reorder algorithm.
Add three images.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2264
- Other reference link(s):

### CODE REVIEW

File: reference/performance/join-reorder.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+title: Introduction to Join Reorder
+summary: Use the Join Reorder algorithm to join multiple tables in TiDB.
+category: reference
+---
+
+# Introduction to Join Reorder
+
+In real business scenarios, it is common to join multiple tables. The execution efficiency of join is associated with the order in which each table joins.
```
Comment (yikeke) [inline reference/performance/join-reorder.md]:
```suggestion
In real application scenarios, it is common to join multiple tables. The execution efficiency of join is associated with the order in which each table joins.
```
“业务场景” often means “应用场景” in TiDB context, so "application" is preferred over "business".

File: reference/performance/join-reorder.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+The above process is the Join Reorder algorithm currently used in TiDB.
+
+## Limitations of Join Reorder algorithm
+
+The current Join Reorder algorithm has the following limitations:
+
+- The Join Reorder for Outer Join is not supported
+- Limited by the calculation methods of the result sets, the algorithm cannot ensure it selects the optimum join order.
+
+Currently, the `STRAIGHT_JOIN` syntax is supported in TiDB to force a join order. For more information, please refer to [Description of the syntax elements](/reference/sql/statements/select.md#description-of-the-syntax-elements).
```
Comment (yikeke) [inline reference/performance/join-reorder.md]:
```suggestion
Currently, the `STRAIGHT_JOIN` syntax is supported in TiDB to force a join order. For more information, refer to [Description of the syntax elements](/reference/sql/statements/select.md#description-of-the-syntax-elements).
```

File: reference/performance/join-reorder.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+category: reference
+---
+
+# Introduction to Join Reorder
+
+In real business scenarios, it is common to join multiple tables. The execution efficiency of join is associated with the order in which each table joins.
+
+For example:
+
+```sql
```
Comment (yikeke) [inline reference/performance/join-reorder.md]:
```suggestion
{{< copyable "sql" >}}

```sql
```

**Comment by ran-huang** [comment]
@winoros PTAL~

**Comment by yikeke** [comment]
ping @winoros

**Pull request merged**